delta-lake/src/main/java/org/apache/iceberg/delta/BaseMigrateDeltaLakeTableAction.java

```diff
@@ -0,0 +1,356 @@
+    if (io != null) {
+      file = io.newInputFile(fullFilePath);
+    } else {
+      file = HadoopInputFile.fromPath(new Path(fullFilePath), this.hadoopConfiguration);
+    }
+
+    MetricsConfig metricsConfig = MetricsConfig.forTable(table);
+    String nameMappingString = table.properties().get(TableProperties.DEFAULT_NAME_MAPPING);
+    NameMapping nameMapping =
+        nameMappingString != null ? NameMappingParser.fromJson(nameMappingString) : null;
+    Metrics metrics = getMetricsForFile(file, format, metricsConfig, nameMapping);
+
+    String partition =
+        spec.fields().stream()
+            .map(PartitionField::name)
+            .map(name -> String.format("%s=%s", name, partitionValues.get(name)))
+            .collect(Collectors.joining("/"));
+
+    return DataFiles.builder(spec)
+        .withPath(fullFilePath)
+        .withFormat(format)
+        .withFileSizeInBytes(file.getLength())
+        .withMetrics(metrics)
+        .withPartitionPath(partition)
+        .build();
+  }
+
+  private FileFormat determineFileFormatFromPath(String path) {
+    if (path.endsWith(PARQUET_SUFFIX)) {
+      return FileFormat.PARQUET;
+    } else if (path.endsWith(AVRO_SUFFIX)) {
+      return FileFormat.AVRO;
+    } else if (path.endsWith(ORC_SUFFIX)) {
+      return FileFormat.ORC;
+    } else {
+      throw new ValidationException("The format of the file %s is unsupported", path);
+    }
+  }
+
+  private Metrics getMetricsForFile(
+      InputFile file, FileFormat format, MetricsConfig metricsSpec, NameMapping mapping) {
+    switch (format) {
+      case AVRO:
+        long rowCount = Avro.rowCount(file);
+        return new Metrics(rowCount, null, null, null, null);
+      case PARQUET:
+        return ParquetUtil.fileMetrics(file, metricsSpec, mapping);
+      case ORC:
+        return OrcMetrics.fromInputFile(file, metricsSpec, mapping);
+      default:
+        throw new ValidationException("Unsupported file format: %s", format);
+    }
+  }
+
+  private static Map<String, String> destTableProperties(
+      io.delta.standalone.Snapshot deltaSnapshot,
+      String originalLocation,
+      Map<String, String> additionalProperties) {
+    Map<String, String> properties = Maps.newHashMap();
+
+    properties.putAll(deltaSnapshot.getMetadata().getConfiguration());
+    properties.putAll(
+        ImmutableMap.of(
+            MIGRATION_SOURCE_PROP, DELTA_SOURCE_VALUE, ORIGINAL_LOCATION_PROP, originalLocation));
+    properties.putAll(additionalProperties);
+
+    return properties;
+  }
+
+  /**
+   * Get the full file path, the input {@code String} path can be either a relative path or an
+   * absolute path of a data file in delta table
+   *
+   * @param path the return value of {@link AddFile#getPath()} or {@link RemoveFile#getPath()}
+   *     (either absolute or relative)
+   * @param tableRoot the root path of the delta table
+   */
+  @VisibleForTesting
+  static String getFullFilePath(String path, String tableRoot) {
```

Comment:
the way this is judging full path or not seems awkward, because the file might not be under table root but still be a full path, especially if it's a migrated table. I think we should check if the file path has a scheme, like `s3://` to determine if this is a valid full path. I believe Hadoop's `Path` class has some ways to check this.

Reply:
Also I don't think we need to specifically test this case. Is there a way we can unit-test this whole base action without relying on a Spark engine?